tests/models/test_linear_network.py

```diff
@@ -10,18 +10,21 @@
 
 class TestLinearNetwork:
 
-    def test_save(self, tmp_path):
+    def test_save(self, tmp_path, monkeypatch):
 
         layer_sizes = [10]
         input_size = 10
         dropout = 0.25
 
-        model = LinearNetwork(data_folder=tmp_path, layer_sizes=layer_sizes,
-                              dropout=dropout)
+        def mocktrain(self):
```

Comment:
how does `mocktrain()` get called? does `mock` prefix mean that whenever you call `.train()` it actually calls `mocktrain()` instead of the true `train()`?